main/src/library/Reducible.flix

```diff
@@ -84,52 +84,62 @@ pub lawless class Reducible[t: Type -> Type] {
     ///
     /// Returns the first element of `t`.
     ///
-    pub def head(t: t[a]): a = ???
+    pub def head(t: t[a]): a =
+        Reducible.reduceLeft((acc, _) -> acc, t)
 
     ///
     /// Returns the last element of `t`.
     ///
-    pub def last(t: t[a]): a = ???
+    pub def last(t: t[a]): a =
+        Reducible.reduceLeft((_, a) -> a, t)
 
     ///
     /// Returns a list with the first element of `t`.
     ///
-    pub def init(t: t[a]): List[a] = ???
+    pub def init(t: t[a]): List[a] =
```

Comment:
without the first

Reply:
Oh. I think there has been a misunderstanding.

`init` returns everything except the first element. Like drop (1).

Reply:
Well isn't that the same semantics as `tail`?

Reply:
I was tired. Its reverse tail reverse.

Reply:
`1 :: 2 :: 3 :: Nil => 2 :: 3 :: Nil` in case I messed up again :)

Reply:
Oh wait I found it in `Nel`:
```flix
///
/// Returns all elements in `l` without the last element.
///
@Time(length(l)) @Space(length(l))
pub def init(l: Nel[a]): List[a] = ...
```